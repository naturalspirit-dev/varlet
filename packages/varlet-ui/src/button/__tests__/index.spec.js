import example from '../example'
import { render, fireEvent } from '@testing-library/vue'

test('test button example', async () => {
  const { html } = render(example)

  expect(html()).toMatchSnapshot()
})

test('test button click', async () => {
  const { getByText } = render(example)
  await fireEvent.click(getByText('点击'))

  expect(getByText('click success!')).toBeTruthy()
})

test('test button touch', async () => {
  const { getByText } = render(example)
  await fireEvent.touchStart(getByText('触摸开始(移动端)'))

  expect(getByText('touchstart success!')).toBeTruthy()
})
