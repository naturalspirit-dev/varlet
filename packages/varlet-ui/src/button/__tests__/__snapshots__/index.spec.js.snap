// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button example 1`] = `"<div class=\\"example\\"><div class=\\"app-type\\">主题色按钮</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\">主要按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--success var-elevation--2\\">成功按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--danger var-elevation--2\\">危险按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--warning var-elevation--2\\">警告按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--info var-elevation--2\\">信息按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--default var-elevation--2\\">默认按钮</button><div class=\\"app-type\\">朴素按钮</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--plain-primary var-button--plain  var-button--outline\\">边框按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--plain-primary var-button--plain\\">文字按钮</button><div class=\\"app-type\\">禁用状态</div><button class=\\"var--box var-button var-button--normal var--inline-flex var-button--disabled var-button--default var-elevation--2\\" disabled=\\"\\">禁用状态</button><button class=\\"var--box var-button var-button--normal var--inline-flex var-button--disabled var-button--plain-primary var-button--plain  var-button--outline\\" disabled=\\"\\">禁用状态</button><button class=\\"var--box var-button var-button--normal var--inline-flex var-button--disabled var-button--plain-primary var-button--plain\\" disabled=\\"\\">禁用状态</button><div class=\\"app-type\\">加载状态</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\"><div class=\\"var--box var-loading\\"><div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div><!--v-if--><!--v-if--><!--v-if--><!--v-if--></div></button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--plain-primary var-button--plain  var-button--outline\\"><div class=\\"var--box var-loading\\"><div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div><!--v-if--><!--v-if--><!--v-if--><!--v-if--></div></button><div class=\\"app-type\\">加载类型</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\"><div class=\\"var--box var-loading\\"><div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 30px; height: 30px;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div><!--v-if--><!--v-if--><!--v-if--><!--v-if--></div></button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\"><div class=\\"var--box var-loading\\"><!--v-if--><!--v-if--><!--v-if--><div class=\\"var-loading__rect var-loading__rect-normal\\"><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div></div><!--v-if--></div></button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\"><div class=\\"var--box var-loading\\"><!--v-if--><!--v-if--><!--v-if--><!--v-if--><div class=\\"var-loading__disappear var-loading__disappear-normal\\"><div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div></div></div></button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\"><div class=\\"var--box var-loading\\"><!--v-if--><!--v-if--><div class=\\"var-loading__cube var-loading__cube-normal\\"><div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div><div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div></div><!--v-if--><!--v-if--></div></button><div class=\\"app-type\\">按钮尺寸</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--primary var-elevation--2\\">常规按钮</button><button class=\\"var--box var-button var-button--small var--inline-flex  var-button--primary var-elevation--2\\">小型按钮</button><button class=\\"var--box var-button var-button--mini var--inline-flex  var-button--primary var-elevation--2\\">迷你按钮</button><button class=\\"var--box var-button var-button--large var--inline-flex  var-button--primary var-elevation--2\\">大型按钮</button><div class=\\"app-type\\">块级flex</div><button class=\\"var--box var-button var-button--normal var--flex  var-button--primary var-elevation--2\\">块级按钮</button><div class=\\"app-type\\">自定义颜色</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--default var-elevation--2\\" style=\\"color: rgb(0, 92, 175);\\">文字颜色</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--default var-elevation--2\\" style=\\"color: rgb(255, 255, 255); background: rgb(156, 39, 176);\\">紫色按钮</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--default var-elevation--2\\" style=\\"color: rgb(255, 255, 255);\\">渐变按钮</button><div class=\\"app-type\\">圆形按钮</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--default var-elevation--2 var-button--round\\"><img style=\\"width: 30px; height: 30px;\\" src=\\"https://img.yzcdn.cn/vant/logo.png\\"></button><div class=\\"app-type\\">注册事件</div><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--danger var-elevation--2\\">点击</button><button class=\\"var--box var-button var-button--normal var--inline-flex  var-button--danger var-elevation--2\\">触摸开始(移动端)</button></div>"`;
